src/controller.js

```diff
@@ -27,38 +35,148 @@ class ChatbotCtrl {
     this.$timeout = $timeout;
     this.$translate = $translate;
     this.ChatbotService = ChatbotService;
+    this.LivechatFactory = LivechatFactory;
+    this.LivechatService = LivechatService;
   }
 
   $onInit() {
+    this.initialized = false;
     this.started = false;
+    this.livechat = false;
     this.hidden = true;
+
     this.messages = [];
     this.suggestions = [];
     this.banner = null;
 
     this.loaders = {
       isStarting: false,
+      isStartingLivechat: false,
       isAsking: false,
       isGettingSuggestions: false,
+      isAnsweringSurvey: false,
+      isSendingSurvey: false,
+      isAgentTyping: false,
+      isLivechatDone: false,
     };
 
     this.options = {
       isEnabled: true,
       showOpenButton: false,
     };
 
+    this.lastLivechatSettings = [];
+
+    this.livechatQuestions = [
+      {
+        id: 'q1',
+        text: 'livechat_survey_question1',
+        answers: [
+          { value: 3, text: 'livechat_survey_answer_over_satisfied', img: 'happy' },
+          { value: 1, text: 'livechat_survey_answer_satisfied', img: 'medium' },
+          { value: 0, text: 'livechat_survey_answer_unsatisfied', img: 'sad' },
+        ],
+      },
+      {
+        id: 'q2',
+        text: 'livechat_survey_question2',
+        answers: [
+          { value: 3, text: 'livechat_survey_answer_over_satisfied', img: 'happy' },
+          { value: 1, text: 'livechat_survey_answer_satisfied', img: 'medium' },
+          { value: 0, text: 'livechat_survey_answer_unsatisfied', img: 'sad' },
+        ],
+      },
+      {
+        id: 'q3',
+        text: 'livechat_survey_question3',
+        answers: [
+          { value: 4, text: 'chatbot_answer_yes', img: 'happy' },
+          { value: 0, text: 'chatbot_answer_no', img: 'sad' },
+        ],
+      },
+    ];
+
     this.$scope.CHATBOT_MESSAGE_QUALITY = CHATBOT_MESSAGE_QUALITY;
     this.$scope.CHATBOT_MESSAGE_TYPES = CHATBOT_MESSAGE_TYPES;
     this.$scope.CHATBOT_SURVEY_STEPS = CHATBOT_SURVEY_STEPS;
 
     this.$rootScope.$on('ovh-chatbot:open', () => this.open());
     this.$rootScope.$on('ovh-chatbot:opened', () => this.focusInput());
+  }
+
+  init() {
```

Comment:
```suggestion
  $onInit() {
```
to be consistent

Reply:
It's actually an init function that should not be called as soon as angular is started, but the first time `open` is called.